js/ui/html_editor/utils/toolbar_helper.js

```diff
@@ -214,52 +221,113 @@ function prepareLinkHandler(module) {
     };
 }
 
-function prepareImageHandler(module) {
+function prepareImageHandler(module, imageUploadOption) {
     return () => {
-        const formData = module.quill.getFormat();
+        let formData = module.quill.getFormat();
         const isUpdateDialog = Object.prototype.hasOwnProperty.call(formData, 'imageSrc');
         const defaultIndex = defaultPasteIndex(module);
 
         if(isUpdateDialog) {
             const { imageSrc } = module.quill.getFormat(defaultIndex - 1, 1);
 
-            formData.src = formData.imageSrc;
-            delete formData.imageSrc;
+            formData = getUpdateDialogFormData(module, formData);
 
             if(!imageSrc || defaultIndex === 0) {
```

Comment:
why it can be `true` if `isUpdateDialog=true` suggests that there is imageSrc?

Reply:
here we check imageSrc for defaultIndex - 1 index to handle the situation when the cursor is before image